src/meta/processors/admin/Balancer.cpp

```diff
@@ -78,14 +90,18 @@ bool Balancer::recovery() {
         CHECK_EQ(1, corruptedPlans.size());
         plan_ = std::make_unique<BalancePlan>(corruptedPlans[0], kv_, client_.get());
         plan_->onFinished_ = [this] () {
-            bool expected = true;
-            auto &running = this->running_;   // Get the reference before the captured `this' lost
-            plan_.reset();
-            CHECK(running.compare_exchange_strong(expected, false));
+            // Get the reference before the captured `this' lost
+            auto &lock = this->lock_;
+            {
+                std::lock_guard<std::mutex> lg(lock);
+                plan_.reset();
+                running_ = false;
```

Comment:
running need captured before this lost too

Reply:
Good catch, I move this line before `plan_.reset()`